src/os/newstore/NewStore.cc

```diff
@@ -3299,8 +3327,8 @@ int NewStore::_do_write_all_overlays(TransContext *txc,
         bl_next_data.substr_of(bl_next, next->second.value_offset,
                                next->second.length);
         bl.claim_append(bl_next_data);
-        txc->t->rmkey(PREFIX_OVERLAY, key_next);
 
+	op->overlays.push_back(next->second);
 	++prev;
 	++next;
       } else {
```

Comment:
below, we need to not clear the shared_overlay keys, as those may be needed.  and we need to put in a wal event that cleans them up (they may be referenced in shared_overlays but not by any of the current overlays we are using because they were once shared and then all because obsolete/overwritten.